external/pmrrr/include/pmrrr/lapack/odnst.hpp

```diff
@@ -0,0 +1,177 @@
+/**
+	C++ template version of LAPACK routine dlanst.
+	Based on C code translated by f2c (version 20061008).
+*/
+
+#ifndef __ODNST_HPP__
+#define __ODNST_HPP__
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <float.h>
+#include <assert.h>
+
+#include <pmrrr/lapack/odssq.hpp>
+#include <pmrrr/lapack/lapack.hpp>
+
+namespace pmrrr { namespace lapack {
+
+	template<typename FloatingType>
```

Comment:
This routine seems like low-hanging fruit for a ground-up implementation as well; also, how will the `fabs` function for non-standard datatypes (e.g., `El::BigFloat`)?